jax/_src/numpy/lax_numpy.py

```diff
@@ -4193,17 +4211,18 @@ def _linspace(start: ArrayLike, stop: ArrayLike, num: int = 50,
     dtype = dtypes.to_inexact_dtype(result_type(start, stop))
   dtype = _jnp_dtype(dtype)
   computation_dtype = dtypes.to_inexact_dtype(dtype)
-  start = asarray(start, dtype=computation_dtype)
-  stop = asarray(stop, dtype=computation_dtype)
+  start = asarray(start, dtype=computation_dtype, device=device)
```

Comment:
Do we need to put device here? can we just add sharding to the final return instead?

Reply:
Also, this might break if `device` is a nontrivial sharding

Reply:
My initial thought was that in eager mode, we'll avoid usage of default device on these internal variables.
If this breaks on a non-trivial shardings, then yes, let's remove it from here.